sklearn/cluster/tests/test_mean_shift.py

```diff
@@ -174,6 +184,7 @@ def test_bin_seeds():
         cluster_std=0.1,
         random_state=0,
     )
+    X = X.astype(global_dtype, copy=False)
```

Comment:
In other place, we used `X_`. Do we want to keep this covention here?

Reply:
I find this convention weird and not very explicit.

If we do not plan to keep the original local variable `X` I find the current code of this PR better.

If we want to keep the original `X`, then let's better use more explicit names, both for the original `X` and the new version.